### PR TITLE
Pluribus Networks port config module

### PR DESCRIPTION
**Pull request by rajaspachipulusu17**
##### SUMMARY

Point feature of port config for Pluribus Networks Netvisor switches 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
pn_port_config.py module with unit tests.

##### ADDITIONAL INFORMATION
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

### CODE REVIEW

**Comment by rajaspachipulusu17** [comment]
@Akasurde can you merge this?

Thanks,
Rajas

**Comment by rajaspachipulusu17** [comment]
Can someone approve? its been quite some time now.

Thanks,
Rajas

**Review by pdam-arista — APPROVED** [review]

File: lib/ansible/modules/network/netvisor/pn_port_config.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+def main():
+    """ This section is for arguments parsing """
+
+    state_map = dict(
+        update='port-config-modify'
+    )
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            pn_cliswitch=dict(required=False, type='str'),
```
Comment (ganeshrn) [inline lib/ansible/modules/network/netvisor/pn_port_config.py:239]: `required=False` is the default value and not required to be passed explicitly. Applies to other places as well.

**Review by ganeshrn — COMMENTED** [review]
Please add an entry for Pluribus network files in https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml with your Github handle as maintainers.

File: lib/ansible/modules/network/netvisor/pn_port_config.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    required: False
+    type: str
+  pn_eth_mode:
+    description:
+      - physical Ethernet mode.
+    required: False
+    choices: ['1000base-x', 'sgmii', 'disabled', 'GMII']
+  pn_fabric_guard:
+    description:
+      - Fabric guard contfiguration.
```
Comment (samccann) [inline lib/ansible/modules/network/netvisor/pn_port_config.py]:
```suggestion
      - Fabric guard configuration.
```

File: lib/ansible/modules/network/netvisor/pn_port_config.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    state: "update"
+    pn_port: "all"
+    pn_dscp_map: "foo"
+
+- name: port config modify
+  pn_port_config:
+    pn_cliswitch: "sw01"
+    state: "update"
+    pn_port: "all"
+    pn_host_enable: True
```
Comment (samccann) [inline lib/ansible/modules/network/netvisor/pn_port_config.py]:
```suggestion
    pn_host_enable: true
```

Comment (samccann) [reply lib/ansible/modules/network/netvisor/pn_port_config.py]: fwiw - the preference is true/false (no capitalization) so it doesn't throw warnings/errors in some linter tools.

**Review by samccann — COMMENTED** [review]
one spelling error and another suggested edits.  Otherwise docs look good.

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
@rajaspachipulusu17 Thank you!

**Pull request merged**